static/js/settings.js

```diff
@@ -0,0 +1,61 @@
+(function() {
+	"use strict";
+
+    // TODO Find a way to make this reusable across the app?
+    function apiRequest(path, options) {
+        const cookieString = document.cookie ?? "";
+        const cookieStringArray = cookieString
+            .split(";")
+            .map(individualCookieString => individualCookieString.split("="))
+            .map(([cookieKey, cookieValue]) => [cookieKey.trim(), cookieValue.trim()]);
+        const [csrfCookieKey, csrfCookieValue] = cookieStringArray.find(([cookieKey, _cookieValue]) => cookieKey === "csrftoken");
+        const headers = new Headers(options?.headers);
+        headers.set("X-CSRFToken", csrfCookieValue);
+        headers.set("Content-Type", "application/json");
+        return fetch(
+            `/api/v1${path}`,
+            {
+                mode: "same-origin",
+                ...options,
+                headers: headers,
+            },
+        );
+    }
+
+    const settingsForm = document.querySelector("#settings-form");
+    const labelCollectionCheckbox = document.querySelector("input#label-collection");
+    const labelCollectionOffWarning = document.querySelector("#label-collection-off-warning");
+    const saveError = document.querySelector("#save-error");
+
+    settingsForm.addEventListener("submit", async (event) => {
+        event.preventDefault();
+        const profileId = settingsForm.dataset.profileId;
+
+        const settings = {
+            server_storage: labelCollectionCheckbox.checked,
+        };
+
+        const response = await apiRequest(
+            `/profiles/${profileId}/`,
+            {
+                method: "PUT",
+                body: JSON.stringify(settings),
+            },
+        );
+        if (response.ok) {
+            // Re-render the page on the server, to make sure all {% if %} statements in the template are rendered correctly:
+            document.location.reload();
```

Comment:
So this is terrible. I wanted to avoid trying to keep the front-end in sync with what the server would render (i.e. manually making the warning at the top appear or disappear or not when the form is submitted depending on the value of the setting), so I just reloaded the entire page to get the server to render it. The disadvantage is that there's no "Saved!" confirmation message (in addition to the page just getting a superfluous reload.)

Reply:
I've ~seen~ written worse. 😉